### ISSUE TITLE
quantile function should be marked const

### ISSUE THREAD

**Issue by BenBE** [sengelha/streaming-percentiles-cpp#32]
The method `stmpct_alg::quantile` and all overriding functions should be marked `const`.

The only place this needs some additional though apart from changing the method's signature is in `double ckms_impl::quantile(double phi)` which also needs the following minor change:

```diff
diff --git a/src/stmpct/ckms_impl.cpp b/src/stmpct/ckms_impl.cpp
index b616460..dae5eda 100644
--- a/cpp/src/ckms_impl.cpp
+++ b/cpp/src/ckms_impl.cpp
@@ -12,7 +12,7 @@ void ckms_impl::insert(double v)
     }
 }
 
-double ckms_impl::quantile(double phi)
+double ckms_impl::quantile(double phi) const
 {
     if (m_S.empty()) {
         return nan("");
@@ -23,13 +23,14 @@ double ckms_impl::quantile(double phi)
     double comp = phin + f(phin, m_n) / 2;
 
     int r_i = 0;
-    tuples_t::iterator it = m_S.begin();
-    assert(it != m_S.end()); // This is guaranteed by the m_S.empty() check above
+    tuples_t::const_iterator it = m_S.cbegin();
+    tuples_t::const_iterator end = m_S.cend();
+    assert(it != end); // This is guaranteed by the m_S.empty() check above
 
-    tuples_t::iterator it_prev = it;
+    tuples_t::const_iterator it_prev = it;
     ++it;
 
-    for (; it != m_S.end(); ++it, ++it_prev) {
+    for (; it != end; ++it, ++it_prev) {
         r_i += it_prev->g;
 
         if (r_i + it->g + it->delta > comp) {
```

Caching the value of `m_S.cend()` in this manner also has the additional upside of avoiding several constructor calls to the underlaying iterator constructor and thus saving some time in the process.

**Comment by sengelha** [comment]
Fixed in commit 757e3df9464005b675fbc55668fda3401ac3a859

**Comment by sengelha** [comment]
Also commit 9c4cf1ceb10329b56300a45023fb831328d94480
